google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -456,6 +478,17 @@ TransactionSelector getTransactionSelector() {
             }
           }
           throw SpannerExceptionFactory.newSpannerException(e.getCause());
+        } catch (TimeoutException e) {
+          SpannerException se =
+              SpannerExceptionFactory.newSpannerException(
+                  ErrorCode.DEADLINE_EXCEEDED,
+                  "Timeout while waiting for a transaction to be returned by another statement. "
+                      + "See the suppressed exception for the stacktrace of the caller that should return a transaction",
+                  e);
+          if (transactionStarter != null) {
+            se.addSuppressed(transactionStarter);
```

Comment:
Include the stacktrace of the statement that should have returned the transaction in the exception that we are throwing.